src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -462,13 +462,14 @@ public async Task AddToRoleAsync(IUser user, string normalizedRoleName, Cancella
             if (user is User u)
             {
                 var roleNames = await _roleService.GetRoleNamesAsync();
+                var roleName = roleNames?.FirstOrDefault(r => NormalizeKey(r) == normalizedRoleName);
 
                 if (!roleNames.Any(r => NormalizeKey(r) == normalizedRoleName))
```

Comment:
if (String.IsNullOrEmpty(roleName))

Reply:
Here we check `RoleNames` not `RoleName`

For now I just copy paste the code as it was before.

Reply:
@jtkech  yes I understand. you don't need to check names all over again since you are doing the same check in the previous line.

The code should look like this if you want to drop `?` since the `GetRoleNameAsync()` would return collection not nullable.

```
        public async Task AddToRoleAsync(IUser user, string normalizedRoleName, CancellationToken cancellationToken)
        {
            if (user == null)
            {
                throw new ArgumentNullException(nameof(user));
            }

            if (user is User u)
            {
                var roleNames = await _roleService.GetRoleNamesAsync();

                var roleName = roleNames.FirstOrDefault(name => NormalizeKey(name) == normalizedRoleName);

                if (String.IsNullOrEmpty(roleName))
                {
                    throw new InvalidOperationException($"Role {normalizedRoleName} does not exist.");
                }

                u.RoleNames.Add(roleName);
            }
        }
```

Reply:
But yes better, done